SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java

```diff
@@ -183,12 +211,48 @@ public void runActionOnPoll() {
             if (isLongRunning(task) ||  isConsideredLongRunning(task)) {
               updateLongRunningTasksUsage(longRunningTasksUsage, latestUsage.getMemoryTotalBytes(), taskCpusUsed, latestUsage.getDiskTotalBytes());
             }
-            SingularityTaskCurrentUsage currentUsage = new SingularityTaskCurrentUsage(latestUsage.getMemoryTotalBytes(), now, taskCpusUsed, latestUsage.getDiskTotalBytes());
+            currentUsage = new SingularityTaskCurrentUsage(latestUsage.getMemoryTotalBytes(), now, taskCpusUsed, latestUsage.getDiskTotalBytes());
 
             usageManager.saveCurrentTaskUsage(taskId, currentUsage);
 
             cpusUsedOnSlave += taskCpusUsed;
           }
+
+          if (slaveOverloaded && configuration.isShuffleTasksForOverloadedSlaves() && currentUsage != null && currentUsage.getCpusUsed() > 0) {
+            if (isLongRunning(task)) {
+              Optional<SingularityTaskHistoryUpdate> maybeCleanupUpdate = taskManager.getTaskHistoryUpdate(task, ExtendedTaskState.TASK_CLEANING);
+              if (maybeCleanupUpdate.isPresent() && isTaskShuffleCleanup(maybeCleanupUpdate.get())) {
+                LOG.trace("Task {} already being cleaned up to spread cpu usage, skipping", taskId);
```

Comment:
nit: I think `isTaskShuffleCleanup` is a little confusing. Maybe `isTaskAlreadyShuffled` or something similar?The trace also implies it's currently being cleaned up, but the method checks the historical statuses so it's possible it was already shuffled. For future debugging it might be good to have the log line note that. E.g. `Task {} is being or was already cleaned up to spread cpu usage, skipping"